File: 2015/ruby/day8/part2.rb
#!/usr/bin/ruby

begin
    File.open('input.txt', 'r') do |file|
        total = 0

        file.each_line do |line|
            string = line[1..-3]
            total += 4
            total += string.count('"\\')
        end

        puts total
    end
rescue SystemCallError
    puts 'input.txt not found.'
end

File: 2016/python/day4/part2.py
#!/usr/bin/python

with open('input.txt') as file_handler:
    content = file_handler.readlines()

for line in content:
    data = line.rstrip(']\n').rsplit('-', 1)
    name = data[0].translate(None, '-')
    sector_id = data[1].split('[')[0]
    name = ''.join(chr((ord(letter) - 97 + int(sector_id)) % 26 + 97) for letter in name)

    if name == 'northpoleobjectstorage':
        print sector_id
        break
